src/dotnet-install.sh

```diff
@@ -713,56 +752,164 @@ download() {
+        product="aspnetcore-runtime"
+    elif [ -z "$runtime" ]; then
+        product="dotnet-sdk"
+    else
+        #impossible combination
+        return 1
+    fi
+    say_verbose "Product to use: '$product'" 
+
+    #quality is not supported for LTS or current channel
+    if [[ ! -z "$quality"  && ("$channel" == "LTS" || "$channel" == "Current") ]]; then
+        quality=""
+        say "Specifying quality for current or LTS channel is not supported, the quality will skipped"
+    fi
+
+    #TODO: check if any other restricions for channel should be applied
+
+    say_verbose "Retrieving primary named payload URL from aka.ms for channel '$channel', quality '$quality' product '$product', os '$osname', architecture '$normalized_architecture'" 
+
+    #construct aka.ms link
+    aka_ms_link="https://aka.ms/dotnet/$channel" 
+    if [[ ! -z "$quality" ]]; then
+        aka_ms_link="$aka_ms_link/$quality"
+    fi
+    aka_ms_link="$aka_ms_link/$product-$osname-$normalized_architecture.tar.gz"
+    say_verbose "Constructed aka.ms link: $aka_ms_link"
+
+    #get HTTP response
+    response="$(get_http_header "$aka_ms_link")"
+
+    say_verbose "Received response: $response"
+    http_code=$( echo "$response" | awk '$1 ~ /^HTTP/ {print $2}' | head -1 )
+
+    #if HTTP code is 301 (Moved Permanently), the redirect link exists
+    if [[ "$http_code" == "301" ]]; then
+        aka_ms_download_link=$( echo "$response" | awk '$1 ~ /^Location/{print $2}' | head -1 )
+        if [[ -z "$aka_ms_download_link" ]]; then
+            say_verbose "The aka.ms link is not valid: failed to get redirect location"
+            return 1
+        fi
+        say_verbose "The redirect location retrieved: $aka_ms_download_link"
+        return 0
+    else
+        say_verbose "The aka.ms link is not valid: received HTTP code: $http_code"
+        return 1
+    fi
+}
+
 calculate_vars() {
     eval $invocation
     valid_legacy_download_link=true
 
     normalized_architecture="$(get_normalized_architecture_from_architecture "$architecture")"
     say_verbose "normalized_architecture=$normalized_architecture"
 
+    #try to get download location from aka.ms link
+    #not applicable when exact version is specified via command or json file
+    if [[ -z "$json_file" && "$version" == "Latest" ]]; then
+
+            valid_aka_ms_link=true;
+            get_download_link_from_aka_ms || valid_aka_ms_link=false
+            
+            if [ "$valid_aka_ms_link" == false ]; then
+                say_verbose "The aka.ms link is not valid: falling back to old approach"
+
+            else
+                say_verbose "Retrieved primary named payload URL from aka.ms link: $aka_ms_download_link"
+                download_link=$aka_ms_download_link
+
+                say_verbose "Attempting legacy download location will be skipped"
+                valid_legacy_download_link=false
+
+                #get version from the path
+                IFS='/'
```

Comment:
I've reused the parsing approach already used in the script. Not sure if it's an optimal solution